spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_target_vm_spec.rb

```diff
@@ -50,25 +81,33 @@
     end
 
     it "should refresh new vm" do
-      vm = FactoryGirl.create(:vm_redhat,
-                              :ext_management_system => @ems,
-                              :uid_ems               => "4f6dd4c3-5241-494f-8afc-f1c67254bf77",
-                              :ems_cluster           => @cluster,
-                              :ems_ref               => "/api/vms/4f6dd4c3-5241-494f-8afc-f1c67254bf77")
+      allow(@ems.ovirt_services).to receive(:cluster_name_href).with(@cluster.ems_ref).and_return(@cluster.name)
+      allow(@ems.refresher).to      receive(:refresh)
 
-      VCR.use_cassette("#{described_class.name.underscore}_target_new_vm") do
-        EmsRefresh.refresh(vm)
+      description = add_vm_event[:description]
+      name        = add_vm_event[:name]
+
+      ep_class = ManageIQ::Providers::Redhat::InfraManager::EventParser
+
+      target_hash, target_klass, target_find = ep_class.parse_new_target(add_vm_event, description, @ems, name)
+
+      new_vm = VCR.use_cassette("#{described_class.name.underscore}_target_new_vm", :allow_unused_http_interactions => true) do
+        EmsRefresh.refresh_new_target(@ems, target_hash, target_klass, target_find)
```

Comment:
Since this uses the new arguments, https://github.com/ManageIQ/manageiq/issues/14247 will have to be merged first for this to be green.
You can check out that branch and test, it passes for me locally.